lib/src/container/imageproxy.rs

```diff
@@ -0,0 +1,137 @@
+//! Run container-image-proxy as a subprocess.
+//! This allows fetching a container image manifest and layers in a streaming fashioni.
+
+use super::{ImageReference, Result};
+use crate::cmdext::CommandRedirectionExt;
+use anyhow::Context;
+use futures_util::{Future, FutureExt, TryFutureExt, TryStreamExt};
+use hyper::body::HttpBody;
+use hyper::client::conn::{Builder, SendRequest};
+use hyper::{Body, Request, StatusCode};
+use std::os::unix::prelude::AsRawFd;
+use std::pin::Pin;
+use std::process::Stdio;
+use tokio::io::{AsyncBufRead, AsyncReadExt};
+
+// What we get from boxing a fallible tokio::spawn() closure.  Note the nested Result.
+type JoinFuture<T> = Pin<Box<dyn Future<Output = Result<Result<T>>>>>;
+
+/// Manage a child process proxy to fetch container images.
+pub(crate) struct ImageProxy {
+    proc: tokio::process::Child,
+    request_sender: SendRequest<Body>,
+    stderr: JoinFuture<String>,
+    driver: JoinFuture<()>,
+}
+
+impl ImageProxy {
+    pub(crate) async fn new(imgref: &ImageReference) -> Result<Self> {
```

Comment:
This is more an issue for https://github.com/cgwalters/container-image-proxy/ right?  So it actually *did* originally work that way, but when I generalized it to support all containers/image backends (e.g. supporting `oci-archive:` and `containers-storage:` and not just a registry) it became much nicer to have it do only one image at a time.

And the reason for that is HTTP is inherently stateless, but interacting with a `containers/image` reference is somewhat stateful (e.g. it will transparently convert an `oci-archive:` to an `oci:` directory in the background in a tempdir).

Reply:
https://github.com/cgwalters/container-image-proxy/commit/74fad3b14853beadf45f7332499c2e12dc6e2d54